cmd/scheduler-builder/cmd/root.go

```diff
@@ -0,0 +1,177 @@
+type PreviousPatches struct {
+	Release            string `yaml:"release"`
+	CherryPickDeadline string `yaml:"cherryPickDeadline"`
+	TargetDate         string `yaml:"targetDate"`
+}
+
+// Schedules struct to define the release schedule for a specific version
+type Schedules struct {
+	Release            string            `yaml:"release"`
+	Next               string            `yaml:"next"`
+	CherryPickDeadline string            `yaml:"cherryPickDeadline"`
+	TargetDate         string            `yaml:"targetDate"`
+	EndOfLifeDate      string            `yaml:"endOfLifeDate"`
+	PreviousPatches    []PreviousPatches `yaml:"previousPatches"`
+}
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:               "scheduler-builder --schedule-path path/to/schedule.yaml",
+	Short:             "scheduler-builder generate a humam readable format of the Kubernetes release schedule",
+	Example:           "scheduler-builder --schedule-path /home/user/kubernetes/sig-release/releases/schedule.yaml",
+	SilenceUsage:      true,
+	SilenceErrors:     true,
+	PersistentPreRunE: initLogging,
+	RunE: func(*cobra.Command, []string) error {
+		return run(opts)
+	},
+}
+
+type options struct {
+	schedulePath string
+	logLevel     string
+}
+
+var opts = &options{}
+
+var (
+	schedulePathFlag = "schedule-path"
+
+	requiredFlags = []string{
+		schedulePathFlag,
+	}
+)
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func init() {
+	rootCmd.PersistentFlags().StringVar(
+		&opts.schedulePath,
+		schedulePathFlag,
+		"",
+		"path where can find the schedule.yaml file",
+	)
+
+	rootCmd.PersistentFlags().StringVar(
+		&opts.logLevel,
+		"log-level",
+		"info",
+		"the logging verbosity, either 'panic', 'fatal', 'error', 'warn', 'warning', 'info', 'debug' or 'trace'",
+	)
+
+	for _, flag := range requiredFlags {
+		if err := rootCmd.MarkPersistentFlagRequired(flag); err != nil {
+			logrus.Fatal(err)
+		}
+	}
+}
+
+func initLogging(*cobra.Command, []string) error {
+	return log.SetupGlobalLogger(opts.logLevel)
+}
+
+func run(opts *options) error {
+	if err := opts.SetAndValidate(); err != nil {
+		return errors.Wrap(err, "validating schedule-path options")
+	}
+
+	logrus.Infof("Reading the schedule file %s...", opts.schedulePath)
+	data, err := ioutil.ReadFile(opts.schedulePath)
+	if err != nil {
+		return errors.Wrap(err, "failed to read the file")
+	}
+
+	var patchSchedule PatchSchedule
+
+	logrus.Info("Parsing the schedule...")
+	err = yaml.UnmarshalStrict(data, &patchSchedule)
+	if err != nil {
+		return errors.Wrap(err, "failed to decode the file")
+	}
+
+	logrus.Info("Generating the markdown output...")
+
+	//nolint:prealloc
+	var output []string
```

Comment:
```suggestion
	output := []string{}
```

and remove the `//nolint:prealloc`